students/alheadstrong/lesson_6/test_mailroom_part4.py

```diff
@@ -0,0 +1,91 @@
+from mailroom_part4 import *
+
+
+def test_1():
+    assert is_name_list('not_list') is False
+
+
+def test_2():
+    assert is_name_list('list') is True
+
+
+def test_3():
+    assert is_name_list('') is False
+
+
+def test_4():
+    assert is_name_existing({'donor_name': 'David Sedaris'}) is True
+
+
+def test_5():
+    assert is_name_existing({'donor_name': 'daVid sedaRiS'}) is True
+
+
+def test_6():
+    assert is_name_existing({'donor_name': 'Amy Sedaris'}) is False
+
+
+def test_7():
+    d = {'donor_name': 'David Sedaris', 'donation': 123.0}
+    add_donation_to_existing(d)
+    assert ddb['David Sedaris'] == [23000, 1200, 2000, 123.0]
+
+
+def test_8():
+    d = {'donor_name': 'New Donor', 'donation': 456.0}
+    add_donation_to_new(d)
+    assert ddb['New Donor'] == [456.0]
+
+
+def test_9():
+    d = {'donor_name': 'New Donor', 'donation': 456.0}
+    letter = ('Dear New Donor,'
```

Comment:
another good pattern is to use names like actual/expected
```
def test_create_letter():
    expected = (...)
    assert create_letter(d) == expected
```
makes it really clear to the reader what the expected result is